doc/book/app_server/luajit_memprof.rst

```diff
@@ -0,0 +1,506 @@
+
+LuaJIT memory profiler
+======================
+
+Stating from version :doc:`2.7.1 </release/2.7.1>`, Tarantool
+has the built-in module called ``memprof`` that implements a LuaJIT memory
+profiler and a profile parser. The profiler provides
+a memory allocation report that helps analyse Lua code and find out the places
+that put the most pressure on the Lua garbage collector.
+
+//?where to put this paragraph//
+Usually developers are not interested in information about allocations
+inside built-ins. So if a Lua built-in function is called from a Lua function,
+all allocations are attributed to this Lua function.
+Otherwise, this event is attributed to a C function.
+
+//?where to put this paragraph//
+Tail call optimization does not create a new call frame, so all allocations
+inside the function called via CALLT/CALLMT bytecodes are attributed to its caller.
+
+.. _profiler_usage:
+
+Working with profiler
+---------------------
+
+Usage of the profiler is two-fold:
+
+1.  :ref:`Collect <profiler_usage_get>` a binary profile of allocations,
+    reallocations, and deallocations in memory related to Lua VM
+    (further, *binary memory profile* or *binary profile* for short).
+2.  :ref:`Parse <profiler_usage_parse>` the binary profile collected and get
+    a human-readable profiling report.
+
+.. _profiler_usage_get:
+
+Collecting binary profile
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To collect a binary profile for a particular part of the Lua code,
+you need to place this part between two ``memprof`` functions,
+namely, ``misc.memprof.start()`` and ``misc.memprof.stop()``, and then execute
+the code under Tarantool.
+
+Below is a piece of simple Lua code named ``test.lua`` to illustrate this.
+
+.. _profiler_usage_example01:
+
+..  code-block:: lua
+    :linenos:
+
+    jit.off()
+    local str, err = misc.memprof.start("memprof_new.bin")
+    -- Lua doesn't create a new frame to call string.rep, and all allocations
+    -- are attributed not to the append() function but to the parent scope.
+    local function append(str, rep)
+        return string.rep(str, rep)
+    end
+
+    local t = {}
+    for _ = 1, 1e5 do
```

Comment:
I suggest to change `_` to `i`, as far as it is used later.
```lua
    for i = 1, 1e5 do
        -- table.insert is the built-in function and all corresponding
        -- allocations are reported in the scope of main chunk.
        table.insert(t,
            append('q', i)
        )
    end
    ```

Reply:
corrected